### PR TITLE
Bug Fix for price and quantity

### LINKED ISSUE #237: [PE-D][Tester A] Pricing
Did not state how many dp should be used. When i type in like 0.0001 it reflects 0.00 to show that the transaction actually happened.Either value not reflected correctly or user guide can explain better or it can be both

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: DarrenCsAcc/ped#4

### LINKED ISSUE #248: [PE-D][Tester A] + Sign in price and quantity
The above example shows a positive integer but it was rejected.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: DarrenCsAcc/ped#7

### PR DESCRIPTION
**Pull request by Jo3LW**
1) Update UG to inform users of rounding off in price 
2) Fix wrong error message for price
3) Fix wrong error message for quantity

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedTransaction.java [hunk 1]
```diff
@@ -91,8 +91,11 @@ public Transaction toModelType() throws IllegalValueException {
...
                     Date.class.getSimpleName()));
         }
-        if (!Date.isValidDate(date)) {
-            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS);
+        if (!Date.isValidDateFormat(date)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS_WRONG_FORMAT);
+        }
+        if (!Date.isValidDateInput(date)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS_WRONG_DATE);
         }
```
Comment (ElginL) [inline src/main/java/seedu/address/storage/JsonAdaptedTransaction.java:99]: What's this? Aren't the first and second if condition the same, which means the second if statement will never run? 🤔

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -197,8 +197,11 @@ public static Quantity parseQuantity(String quantity) throws ParseException {
     public static Date parseDate(String date) throws ParseException {
         requireNonNull(date);
         String trimmedDate = date.trim();
-        if (!Date.isValidDate(trimmedDate)) {
-            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+        if (!Date.isValidDateFormat(trimmedDate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS_WRONG_FORMAT);
+        }
+        if (!Date.isValidDateInput(trimmedDate)) {
+            throw new ParseException(Date.MESSAGE_CONSTRAINTS_WRONG_DATE);
```
Comment (ElginL) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:204]: Both conditions in both if statements are the same, how would the 2nd one ever run?

**Review by ElginL — APPROVED** [review]
LGTM

**Review by ElginL — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -197,8 +197,11 @@ public static Quantity parseQuantity(String quantity) throws ParseException {
     public static Date parseDate(String date) throws ParseException {
         requireNonNull(date);
         String trimmedDate = date.trim();
-        if (!Date.isValidDate(trimmedDate)) {
-            throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+        if (!Date.isValidDateFormat(trimmedDate)) {
```
Comment (quannam0124) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:200]: Great use of conditionals!

**Review by jicsontoh — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/model/transaction/DateTest.java [hunk 4]
```diff
@@ -27,26 +27,34 @@ public void constructor_invalidName_throwsIllegalArgumentException() {
...
+
 
         // valid date
-        assertTrue(Date.isValidDate("11/09/2000")); // integers only
-        assertTrue(Date.isValidDate("29/02/2004")); // leap year
+        assertTrue(Date.isValidDateFormat("11/09/2000")); // integers only
+        assertTrue(Date.isValidDateFormat("29/02/2004")); // leap year
     }
 
+    @Test
```
Comment (quannam0124) [inline src/test/java/seedu/address/model/transaction/DateTest.java:43]: Nice example dates

**Review by quannam0124 — APPROVED** [review]
LGTM

**Pull request merged**